one-ui/src/main/java/com/marklogic/hub/curation/controllers/AbstractArtifactController.java

```diff
@@ -1,31 +1,49 @@
 package com.marklogic.hub.curation.controllers;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.marklogic.client.DatabaseClient;
 import com.marklogic.hub.dataservices.ArtifactService;
 import com.marklogic.hub.oneui.models.HubConfigSession;
+import com.marklogic.hub.oneui.services.EnvironmentService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
 public abstract class AbstractArtifactController {
     @Autowired
     HubConfigSession hubConfig;
 
+    @Autowired
+    EnvironmentService environmentService;
+
+    protected ArrayNode allArtifactTypeInfo = null;
+
     protected ResponseEntity<JsonNode> getArtifacts() {
         return new ResponseEntity<>(getArtifactService().getList(this.getArtifactType()), HttpStatus.OK);
     }
 
-    protected ResponseEntity<JsonNode> updateArtifact(String artifactName, JsonNode loadDataJson) {
-        return new ResponseEntity<>(getArtifactService().setArtifact(this.getArtifactType(), artifactName, loadDataJson), HttpStatus.OK);
+    protected ResponseEntity<JsonNode> updateArtifact(String artifactName, JsonNode artifactJson) throws IOException {
+        ResponseEntity<JsonNode> resp = new ResponseEntity<>(getArtifactService().setArtifact(this.getArtifactType(), artifactName, artifactJson), HttpStatus.OK);
```

Comment:
This strikes me as something that would be in an ArtifactManager class that depends on both HubProject and ArtifactService. The controller ideally only has one dependency and only calls a single method; its job is then just to collect inputs from the HTTP request and convert those into a single method call, and then convert whatever the output is into something in the HTTP response. 

That allows the ArtifactManager to be used in any context and keeps the web tier as thin as possible. Is that possible here?

Reply:
It is definitely possible to move it to an Artifact Manager. I thought we wanted to move away from the manager approach, but I guess by having the Artifact Manager we can eventually deprecate the other managers and reduce the overhead of the all the managers we maintain that way.

Reply:
I think the pattern of a Manager having dependencies on both ML and the filesystem makes sense, while a Service only depends on ML (and the Service class is code-generated based on a data service). In some use cases, a Controller may only need to talk to a Service. In probably most use cases for now, a Controller will talk to a Manager who knows how to handle operations that involve both ML and the filesystem.

Reply:
updated to use an artifact manager